### PR TITLE
Fix traceback when git is not installed

### LINKED ISSUE #2809: `rose app-run` fails if git is not installed
```
$ ./test-app-run 
+ cat rose-app.conf
[command]
default=echo Hello
+ rose app-run
Traceback (most recent call last):
  File "..../python3.9/site-packages/metomi/rose/popen.py", line 267, in run_bg
    proc = Popen(args, text=True, **kwargs)
  File "..../python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "..../python3.9/subprocess.py", line 1837, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'git'
 
During handling of the above exception, another exception occurred:
 
Traceback (most recent call last):
  File "/data/local/fcm/cylc-8.3.3-2/bin/rose", line 10, in <module>
    sys.exit(rose())
  File "..../python3.9/site-packages/metomi/rose/rose.py", line 281, in rose
    main(
  File "..../python3.9/site-packages/metomi/rose/rose.py", line 396, in main
    exec_sub_cmd(ns, sub_cmd, cmd_args)
  File "..../python3.9/site-packages/metomi/rose/rose.py", line 194, in exec_sub_cmd
    _exec_python(
  File "..../python3.9/site-packages/metomi/rose/rose.py", line 230, in _exec_python
    fcn()
  File "..../python3.9/site-packages/metomi/rose/app_run.py", line 553, in main
    runner = AppRunner(event_handler)
  File "..../python3.9/site-packages/metomi/rose/app_run.py", line 370, in __init__
    Runner.__init__(self, *args, **kwargs)
  File "..../python3.9/site-packages/metomi/rose/run.py", line 125, in __init__
    config_pm = ConfigProcessorsManager(event_handler, popen, fs_util)
  File "..../python3.9/site-packages/metomi/rose/config_processor.py", line 109, in __init__
    SchemeHandlersManager.__init__(
  File "..../python3.9/site-packages/metomi/rose/scheme_handler.py", line 98, in __init__
    handler = class_(*args, **kwargs)
  File "..../python3.9/site-packages/metomi/rose/config_processors/fileinstall.py", line 65, in __init__
    self.loc_handlers_manager = PullableLocHandlersManager(
  File "..../python3.9/site-packages/metomi/rose/config_processors/fileinstall.py", line 868, in __init__
    SchemeHandlersManager.__init__(
  File "..../python3.9/site-packages/metomi/rose/scheme_handler.py", line 98, in __init__
    handler = class_(*args, **kwargs)
  File "..../python3.9/site-packages/metomi/rose/loc_handlers/git.py", line 39, in __init__
    ret_code, versiontext, stderr = self.manager.popen.run(
  File "..../python3.9/site-packages/metomi/rose/popen.py", line 234, in run
    proc = self.run_bg(*args, **kwargs)
  File "..../python3.9/site-packages/metomi/rose/popen.py", line 271, in run_bg
    raise RosePopenError(args, 1, "", str(exc))
metomi.rose.popen.RosePopenError: git version # return-code=1, stderr=
[Errno 2] No such file or directory: 'git'
```

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #2809 - fixes `rose app-run` and `rose task-run` crashing if git is not installed

### CODE REVIEW

**Comment by MetRonnie** [comment]
- Python 3.7 test is timing out which seems to happen flakily, not unique to this branch
- MacOS test is failing on 8.3.x

**Comment by MetRonnie** [comment]
Do you think it needs a 2nd reviewer?

File: metomi/rose/popen.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ def run_bg(self, *args, **kwargs):
         except OSError as exc:
             if exc.filename is None and args:
                 exc.filename = args[0]
-            raise RosePopenError(args, 1, "", str(exc))
+            raise RosePopenError(args, exc.errno, "", str(exc)) from None
```
Comment (oliver-sanders) [inline metomi/rose/popen.py:271]: Python's errnos are between 1 and 132, so this should be fine.

**Review by oliver-sanders — APPROVED** [review]

**Comment by oliver-sanders** [comment]
(poked tests)

**Comment by oliver-sanders** [comment]
> Hmm, the Python 3.7 tests seem to be consistently hanging 😢 But not for me locally with a fresh Python 3.7 environment

It looks like that is also happening on master.

https://github.com/metomi/rose/actions/runs/11587156748

**Pull request merged**